pkg/handlers/aws/mutation/instancetype/inject_control_plane.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2023 D2iQ, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package instancetype
+
+import (
+	"context"
+	_ "embed"
+
+	"github.com/d2iq-labs/capi-runtime-extensions/pkg/handlers"
+
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	runtimehooksv1 "sigs.k8s.io/cluster-api/exp/runtime/hooks/api/v1alpha1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/d2iq-labs/capi-runtime-extensions/api/v1alpha1"
+	"github.com/d2iq-labs/capi-runtime-extensions/common/pkg/capi/apis"
+	commonhandlers "github.com/d2iq-labs/capi-runtime-extensions/common/pkg/capi/clustertopology/handlers"
+	"github.com/d2iq-labs/capi-runtime-extensions/common/pkg/capi/clustertopology/handlers/mutation"
+	"github.com/d2iq-labs/capi-runtime-extensions/common/pkg/capi/clustertopology/patches"
+	"github.com/d2iq-labs/capi-runtime-extensions/common/pkg/capi/clustertopology/patches/selectors"
+	"github.com/d2iq-labs/capi-runtime-extensions/common/pkg/capi/clustertopology/variables"
+	capav1 "github.com/d2iq-labs/capi-runtime-extensions/common/pkg/external/sigs.k8s.io/cluster-api-provider-aws/v2/api/v1beta2"
+	awsclusterconfig "github.com/d2iq-labs/capi-runtime-extensions/pkg/handlers/aws/clusterconfig"
+	"github.com/d2iq-labs/capi-runtime-extensions/pkg/handlers/generic/clusterconfig"
+)
+
+const (
+	// VariableName is the external patch variable name.
+	VariableName = "instanceType"
+
+	// WorkerHandlerNamePatch is the name of the inject handler.
+	ControlPlaneHandlerNamePatch = "AWSInstanceTypeControlPlanePatch"
+)
+
+type awsInstanceTypeControlPlanePatchHandler struct {
+	variableName      string
+	variableFieldPath []string
+}
+
+var (
+	_ commonhandlers.Named     = &awsInstanceTypeControlPlanePatchHandler{}
+	_ mutation.GeneratePatches = &awsInstanceTypeControlPlanePatchHandler{}
+	_ mutation.MetaMutator     = &awsInstanceTypeControlPlanePatchHandler{}
+)
+
+func NewControlPlaneMetaPatch() *awsInstanceTypeControlPlanePatchHandler {
+	return newAWSInstanceTypeControlPlanePatchHandler(
+		clusterconfig.MetaVariableName,
+		clusterconfig.MetaControlPlaneConfigName,
+		awsclusterconfig.AWSVariableName,
+		VariableName,
+	)
+}
+
+func newAWSInstanceTypeControlPlanePatchHandler(
+	variableName string,
```

Comment:
```suggestion
	metaVariableName string,
```
explicitly using `meta` in the name helps understanding the code better.

Reply:
We use the "meta" prefix to indicate that its like `clusterConfig` or `workerConfig`. Lets instead refactor all of it and drop this concept of "meta" handlers from all handlers.

Reply:
thats sounds good as well. Lets pair on it. I also want to propose if we should consider replacing `meta`, `mutate`, `inject` keywords